Analysis/interface/TauSelection.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "TauMLTools/Analysis/interface/GenLepton.h"
+#include "TauMLTools/Analysis/interface/AnalysisTypes.h"
+#include "TauMLTools/Core/interface/AnalysisMath.h"
+#include "TauMLTools/Core/interface/exception.h"
+
+namespace analysis {
+    boost::optional<GenLeptonMatch> GetGenLeptonMatch(const tau_tuple::Tau& tau)
+    {   
+        using GTGenLeptonKind = reco_tau::gen_truth::GenLepton::Kind;
+        const GTGenLeptonKind genLepton_kind = static_cast<GTGenLeptonKind> (tau.genLepton_kind);
+        GenLeptonMatch match;
+
+        if (tau.genLepton_index >= 0){
+            LorentzVectorM genv(tau.tau_pt, tau.tau_eta, tau.tau_phi, tau.tau_mass);
+            LorentzVectorM tauv(  tau.genLepton_vis_pt  ,
+                                tau.genLepton_vis_eta ,
+                                tau.genLepton_vis_phi ,
+                                tau.genLepton_vis_mass);
+
+            if (ROOT::Math::VectorUtil::DeltaR(tauv, genv) > 0.2){
+                return boost::none;
+            }
+
+            if (genLepton_kind == GTGenLeptonKind::PromptElectron){
+                if (tau.genLepton_vis_pt < 8.0) return boost::none;
+                match = GenLeptonMatch::Electron;
```

Comment:
you are mixing 'match = ...' and `return`. Maybe better always use `return`  and remove `GenLeptonMatch match`?

Reply:
done in https://github.com/cms-tau-pog/TauMLTools/pull/40/commits/7762e783ce87d67576c9c4e6f8f2b957b57f0858